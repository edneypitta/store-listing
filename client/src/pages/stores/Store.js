import React from 'react'
import { getAvailability } from '../../selectors/schedule'
import Tag from './Tag'

const Store = ({ store, tagFilters, onTagClick }) => {
  return (
    <div className='store'>
      <h2>{store.name}</h2>
      <p>{store.description}</p>
      <span>{getAvailability(store.schedule)}</span>
      <div className='tags'>
        {store.tags.map(tag =>
          <Tag key={tag} tag={tag} tagFilters={tagFilters} onTagClick={onTagClick} />)}
      </div>
    </div>
  )
}

export default Store
