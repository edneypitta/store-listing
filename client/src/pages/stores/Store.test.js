import React from 'react'
import { render } from 'enzyme'
import Store from './Store'

describe('Store', () => {
  it('renders without crashing', () => {
    const store = { tags: ['fat'] }
    const component = render(
      <Store store={store} tagFilters={[]} />)

    expect(component).toMatchSnapshot()
  })
})

