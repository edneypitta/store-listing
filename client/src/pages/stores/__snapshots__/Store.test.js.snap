// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store renders without crashing 1`] = `
<div
  class="store"
>
  <h2 />
  <p />
  <span>
    Unknown opening times
  </span>
  <div
    class="tags"
  >
    <button
      class=""
      title="Click to filter by tag"
      value="fat"
    >
      <span
        class="undefined"
      >
        <span
          style="width:1em;height:1em;margin:0 .05em 0 .1em;vertical-align:-0.1em"
        >
          üç©
        </span>
      </span>
      <span
        class="tag-name"
      >
        fat
      </span>
    </button>
  </div>
</div>
`;
